### PR TITLE
Add action to check copyright [ci/bernede1/copyright-2021]

### PR DESCRIPTION
**Pull request by adrienbernede**
A simple action to test if copyrights are up-to-date after #2021.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2037](https://github.com/mfem/mfem/pull/2037) | @adrienbernede | @tzanio | @tzanio + @acfisher + @v-dobrev | 02/03/21 | 02/04/21 | 02/04/21 | |

### CODE REVIEW

**Comment by adrienbernede** [comment]
The action was meant to fail to test it. Now pushing the fix.

**Comment by adrienbernede** [comment]
@tzanio that seems ready.

File: .github/workflows/repo-check.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  copyright-check:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: checkout MFEM
```
Comment (tzanio) [inline .github/workflows/repo-check.yml:27]: Why the capitals, can't this be just `mfem`?

File: .github/workflows/repo-check.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      with:
+        path: MFEM
+
+    - name: copyright check
+      run: |
+        cd MFEM
+        if git grep -l "^#.*\-2020" > matches.txt
+        then
+          echo "Some files with outdated copyrights were found."
+          echo "Please update to 2021. Here is the list:"
```
Comment (tzanio) [inline .github/workflows/repo-check.yml:38]:
```suggestion
          echo "Please update the following files to Copyright (c) 2010-2021:"
```

**Comment by tzanio** [comment]
Thanks @adrienbernede!

This looks good but it got me thinking about similar checks we do with the tests scripts, https://github.com/mfem/mfem/tree/master/tests/scripts.

Two questions about that:

1. Do you think we should add this as a script so it could also be run locally?

2. How will the currents scripts, gitignore, code-style, documentation, branch-history be handled in GH actions? Is it worth adding them in the same framework as part of this PR?

File: .github/workflows/repo-check.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+    - name: checkout MFEM
+      uses: actions/checkout@v2
+      with:
+        path: MFEM
+
+    - name: copyright check
+      run: |
+        cd MFEM
+        if git grep -l "^#.*\-2020" > matches.txt
```
Comment (tzanio) [inline .github/workflows/repo-check.yml:35]: nice

**Comment by adrienbernede** [comment]
> Thanks @adrienbernede!
> 
> This looks good but it got me thinking about similar checks we do with the tests scripts, https://github.com/mfem/mfem/tree/master/tests/scripts.
> 
> Two questions about that:
> 
> 1. Do you think we should add this as a script so it could also be run locally?

I don't think so, unless you think this will happen often. With all the files having the correct copyright, we can hope that any new PR, even with new files, will have "2021".
For existing PR, yes there may be some new files to fix, but that will be quickly caught by the action, and quickly fixed.

> 2. How will the currents scripts, gitignore, code-style, documentation, branch-history be handled in GH actions? Is it worth adding them in the same framework as part of this PR?

I named the workflow "repo-check" because I want to add the other checks there, indeed. Once this is merged into master, I will update my github-actions branch and add the checks. That way we don't wait and can have the copyright check in master now.

File: .github/workflows/repo-check.yml (hunk 1 above)
Comment (adrienbernede) [reply .github/workflows/repo-check.yml:27]:
```suggestion
    - name: checkout mfem
```

File: .github/workflows/repo-check.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+jobs:
+  copyright-check:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: checkout mfem
+      uses: actions/checkout@v2
+      with:
+        path: MFEM
```
Comment (tzanio) [inline .github/workflows/repo-check.yml:30]: It is minor, but can these also be `mfem`?

**Review by tzanio — APPROVED** [review]

**Review by acfisher — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**